### ISSUE TITLE
Option to 

### ISSUE THREAD

**Issue by dwertheimer** [NotePlan/plugins#550]
**Checks**
- [ ] I confirm I have restarted NotePlan and this problem still persists.
- [ ] Is this a repeatable problem?

**Describe the bug**

What do you expect it to do instead?

**To Reproduce**

**Screenshots**

**System**
 - Device: [e.g. iPhone13, Mac]
 - OS: [e.g. iOS8.1, macOS 10.15]
 - NotePlan Version [e.g. 3.0.25]
 - Plugin Name & Version [e.g. "Templates" v0.2]

**Additional context**

**Plugin Console Log**
To provide us with more clues about where the bug/error is occurring...
1. Open the Plugin's Preferences by going to `NotePlan's menu > Preferences > Plugins` and clicking the settings "cog" icon next to the plugin in question. Scroll to the bottom and set the logging level to "DEBUG" and click "Save & Close"
3. Now open the Plugin Console by going to `Noteplan > Help > Plugin Console` (_not_ the macOS Console app).
4. Run the plugin command you're reporting
5. Copy the output from the Plugin Console and paste it below
6. Delete any output that has personal information you don't want in there
```javascript
    PASTE_PLUGIN_CONSOLE_OUTPUT_HERE
```

**Comment by jgclark** [comment]
Agreed. Once we've moved to React.

**Comment by dwertheimer** [comment]
@jgclark I realized that there's also a bug caused by the multiple renders of the same task in the today section and the #home section. When you click "done" on the task in the top section, it fades and goes away but the one in #home stays there (until you refresh). This can easily be addressed in React, but only if you realize that this task is in multiple sections and needs to be removed from the data structure in multiple sections. Something to think about.

**Comment by jgclark** [comment]
Interesting point, @dwertheimer.
It makes me wonder whether a better (and faster) approach to the duplicate
problem is to accept that both copies show in their respective sections,
but instead get smarter about removing all copies when an item is closed.
How does that sound?

**Comment by dwertheimer** [comment]
@jgclark Yes we should make them both disappear, but I'd like to have a setting (that can be toggled with a keystroke like your other settings) that hides it from all sections except the tag-focused section. And I am willing to do the coding and PR it since it's important to me but clearly not you. My use case is this: I tag things #home that I want to do when I am at home. It's often 4-5 things during the day. I like being able to see those things when I am actually home. But for 50% of the day, it's clutter in my todo list that I constantly have to look past. So for me, I think i would be a lot better to be able to hide/show those items. So in my case, my preference would be to have them *not* in the today section, but underneath the #home section (ideally collapsed until I click to expand it). Again, I'm happy to do the coding if you're open to it.

**Comment by jgclark** [comment]
Aha. In which case I think this is part of a different requirement again,
and one which I'd got on my long list to think about. I think there's a
case to be made to have NP operate in different 'modes' or 'zones' (or
better wording), including the Dashboard.

There's already the ability to ignore certain folders, and to "Ignore items
in calendar sections with this phrase", but with a little more scaffolding
this could be made more comprehensive, with toggles to make it very easy to
shift between the different 'modes'.  I suspect it will need a separate
HTML view to manage the settings for it.

Are we thinking along similar lines?

**Comment by dwertheimer** [comment]
@jgclark Ah that's interesting. An easy way to do what you describe would be:
a) have a setting in plugin settings for "Quick Filters" and in my case I might put "@home,#clientA,#clientB,supermarket" (then we don't need to build an HTML page to manage it)
b) on the dashboard we have a pulldown for Quick Filters, and if I select @home, it shows me only tasks that match this string. Super easy to do in React. No page refreshes required. 

Note that the last example is a simple string "supermarket" showing that you could use any text string, does not have to be a # or @

**Comment by dwertheimer** [comment]
Maybe something like this: https://github.com/weblineindia/ReactJS-Multiselect-Dropdown?tab=readme-ov-file#demo

**Comment by jgclark** [comment]
I think perhaps the word I'm after is having multiple **Perspectives** than we can select between.  Each would have what you describe above, plus I was thinking of going further and letting the user hide/show certain sections.

**Comment by jgclark** [comment]
Another thought has occurred to me. I'm keen that this 'Perspectives' requirement can apply to both Dashboard and Projects. Why? Because this feels particularly relevant to users like George C and myself who are managing >100 projects/areas, and large collections of tasks.  It would be great to be able to switch perspectives more 'centrally' and for it to be picked up by a number of relevant plugins.

So, @dwertheimer, while I appreciate your implementation suggestions, I'm trying to hold back on that, and discuss the requirement in more detail first.  We really want to be able to write the text now that will go at the top of the README summarising in one sentence what this feature offers users.
